### PR TITLE
🐛 Source Facebook Marketing: fix url parsing and add report that exposes conversions.

### PR DESCRIPTION
**Pull request by manavkohli**
## What
This solves the issue of being able to fetch conversion data from the Facebook Marketing API. The discussion can be found here: https://github.com/airbytehq/airbyte/issues/5190

## How
This pull request creates a new report that fetches the `action_type` (where `omni_purchase`) represents a conversion whereas the other reports cannot due to misconfigured `breakdown`s

## Recommended reading order
1. `x.java`
2. `y.python`

#### Community member or Airbyter
   
- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

### CODE REVIEW

**Comment by sherifnada** [comment]
@manavkohli thanks! will review it soon

**Comment by manavkohli** [comment]
Awesome, thanks!

**Comment by manavkohli** [comment]
@sherifnada @vitaliizazmic any feedback here?

**Comment by manavkohli** [comment]
Test results:
```
./gradlew :airbyte-integrations:connectors:source-facebook-marketing:unitTest
**         =============================== warnings summary ===============================
         .venv/lib/python3.9/site-packages/jsonschema/compat.py:6
         .venv/lib/python3.9/site-packages/jsonschema/compat.py:6
           /Users/manavkohli/dev/airbyte/airbyte-integrations/connectors/source-facebook-marketing/.venv/lib/python3.9/site-packages/jsonschema/compat.py:6: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
             from collections import MutableMapping, Sequence  # noqa
         
         -- Docs: https://docs.pytest.org/en/stable/warnings.html
         ======================== 6 passed, 2 warnings in 15.28s ========================
**
```
```

@sherifnada do I need a particular pair of keys to test the integration tests? They're failing for me on master.

```

**Comment by sherifnada** [comment]
@manavkohli to run those tests you need to add credentials for your connector in `you need to add a config.json conforming to the connector’s spec in the facebook marketing module airbyte-integrations/connectors/source-facebook-marketing/secrets/config.json` - would you be able to do that?

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py [hunk 1]
```diff
@@ -47,15 +47,12 @@
...
-    parsed_url = urlparse.urlparse(url)
-    res_query = []
-    for q in parsed_url.query.split("&"):
-        key, value = q.split("=")
-        if key not in params:
-            res_query.append(f"{key}={value}")
-
-    parse_result = parsed_url._replace(query="&".join(res_query))
-    return urlparse.urlunparse(parse_result)
+    parsed = urlparse.urlparse(url)
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py:50]: would you mind adding a unit test for this method?

**Review by sherifnada — COMMENTED** [review]
LGTM! one question on unit tests

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py (hunk 1 above)
Comment (manavkohli) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py:50]: Yes definitely!

**Comment by manavkohli** [comment]
@sherifnada I actually did, but it looks like the tests are written to assert for a specific account, e.g:

```
  @pytest.mark.parametrize("stream_name, deleted_num", [("ads", 2), ("campaigns", 3), ("ad_sets", 1)])
             def test_streams_with_include_deleted_and_state(self, stream_name, deleted_num, config_with_include_deleted, configured_catalog, state):
                 """Should ignore state because of include_deleted enabled"""
                 catalog = self.slice_catalog(configured_catalog, {stream_name})
                 records, states = self._read_records(config_with_include_deleted, catalog, state=state)
                 deleted_records = list(filter(self._deleted_record, records))
             
         >       assert len(deleted_records) == deleted_num, f"{stream_name} should have {deleted_num} deleted records returned"
         E       AssertionError: ads should have 2 deleted records returned
         E       assert 10 == 2
         E         +10
         E         -2

def test_streams_with_include_deleted(self, stream_name, deleted_id, config_with_include_deleted, configured_catalog):
                 catalog = self.slice_catalog(configured_catalog, {stream_name})
                 records, states = self._read_records(config_with_include_deleted, catalog)
                 deleted_records = list(filter(self._deleted_record, records))
                 is_specific_deleted_pulled = deleted_id in list(map(self._object_id, records))
             
                 assert states, "incremental read should produce states"
                 for name, state in states[-1].state.data.items():
                     assert "include_deleted" in state, f"State for {name} should include `include_deleted` flag"
             
                 assert deleted_records, f"{stream_name} stream should have deleted records returned"
         >       assert is_specific_deleted_pulled, f"{stream_name} stream should have a deleted record with id={deleted_id}"
         E       AssertionError: ad_sets stream should have a deleted record with id=23846541706990398
         E       assert False
```

It looks like both of those tests are dependent on data from a particular account?

**Comment by manavkohli** [comment]
Added test coverage results:
```
         =============================== warnings summary ===============================
         .venv/lib/python3.9/site-packages/jsonschema/compat.py:6
         .venv/lib/python3.9/site-packages/jsonschema/compat.py:6
           /Users/manavkohli/dev/airbyte/airbyte-integrations/connectors/source-facebook-marketing/.venv/lib/python3.9/site-packages/jsonschema/compat.py:6: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
             from collections import MutableMapping, Sequence  # noqa
         
         -- Docs: https://docs.pytest.org/en/stable/warnings.html
         ======================= 14 passed, 2 warnings in 15.26s ========================
```

**Comment by sherifnada** [comment]
> It looks like both of those tests are dependent on data from a particular account?

Ah, it's a hack we've done in the test for FB to get around some limitations of our test infra. If you change that ID to the ID of your ad account do the tests pass? if so could you post a screenshot of the passing tests? we should be good to go aftewards

**Review by keu — APPROVED** [review]

**Comment by sherifnada** [comment]
@manavkohli we've been debugging some test failures on our side. I'm pretty sure they should be non-blocking for your PR so I just published the connector and @vitaliizazmic will create an issue to follow up on the test failures. Thanks for your contribution!

**Pull request merged**